### PR TITLE
HPCC-21847 Compile Roxie with all warnings as errors 

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
Compiled. There should be no effect on functionality.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Split into two commits for ease of review - first just fixes the warning related to evaluation order of initializers, second fixes a few other warnings

**Comment by AttilaVamos** [comment]
@richardkchapman 
I think this is false positive:
/mnt/disk1/home/vamosax/smoketest/PR-12383/HPCC-Platform/dali/ft/filecopy.hpp:124:21: error: \xe2\x80\x98DfuOperatonStr\xe2\x80\x99 defined but not used [-Werror=unused-variable]

because the DfuOperatonSt[] is used in FileSparyer::getOperationTypeString() which is used to generate file History record.

**Comment by richardkchapman** [comment]
It's not a false positive, because other cpp files that include this header file do not use the variable. Since it's marked static every cpp file's copy of it is independent.

Hopefully my latest commit should address it.

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: roxie/ccd/ccdcontext.cpp [hunk 1]
```diff
@@ -2637,7 +2637,7 @@ class CRoxieServerDebugContext : extends CBaseServerDebugContext
                 StringBuffer slaveActivityId;
                 slaveInfo.read(slaveActivityId);
                 IActivityDebugContext *slaveActivityCtx = slaveActivityId.length() ? currentGraph->lookupActivityByEdgeId(slaveActivityId.str()) : NULL;
-                BreakpointActionMode action = checkBreakpoint(slaveState, slaveActivityCtx , NULL);
+               checkBreakpoint(slaveState, slaveActivityCtx , NULL);
```
Comment (ghalliday) [inline roxie/ccd/ccdcontext.cpp]: minor: indentation

File: roxie/ccd/ccdfile.cpp [hunk 2]
```diff
@@ -1900,17 +1900,17 @@ class CResolvedFile : implements IResolvedFileCreator, implements ISDSSubscripti
     IMPLEMENT_IINTERFACE;
 
     CResolvedFile(const char *_lfn, const char *_physicalName, IDistributedFile *_dFile, RoxieFileType _fileType, IRoxieDaliHelper* _daliHelper, bool isDynamic, bool cacheIt, bool writeAccess, bool _isSuperFile)
-    : daliHelper(_daliHelper), lfn(_lfn), physicalName(_physicalName), dFile(_dFile), fileType(_fileType), isSuper(_isSuperFile)
+    : lfn(_lfn), physicalName(_physicalName), dFile(_dFile), fileType(_fileType), isSuper(_isSuperFile), daliHelper(_daliHelper)
     {
         cached = NULL;
         fileSize = 0;
-        fileCheckSum = 0;
+        //fileCheckSum = 0;
```
Comment (ghalliday) [inline roxie/ccd/ccdfile.cpp]: Is this intentional?

File: roxie/ccd/ccdfile.cpp [hunk 3]
```diff
@@ -1900,17 +1900,17 @@ class CResolvedFile : implements IResolvedFileCreator, implements ISDSSubscripti
...
         if (dFile)
         {
             if (traceLevel > 5)
                 DBGLOG("Roxie server adding information for file %s", lfn.get());
             bool tsSet = dFile->getModificationTime(fileTimeStamp);
-            bool csSet = dFile->getFileCheckSum(fileCheckSum);
+            //bool csSet = dFile->getFileCheckSum(fileCheckSum);
```
Comment (ghalliday) [inline roxie/ccd/ccdfile.cpp]: is this intentional?

File: roxie/ccd/ccdmain.cpp [hunk 4]
```diff
@@ -351,10 +351,8 @@ static void roxie_common_usage(const char * progName)
 
 class MAbortHandler : implements IExceptionHandler
 {
-    unsigned dummy; // to avoid complaints about an empty class...
+//    unsigned dummy = 0; // to avoid complaints about an empty class...
```
Comment (ghalliday) [inline roxie/ccd/ccdmain.cpp]: remove completely...

File: system/security/shared/seclib.hpp [hunk 5]
```diff
@@ -55,9 +55,9 @@ enum SecAccessFlags : int
     SecAccess_Full = 255
 };
 
-static const char * getSecAccessFlagName(SecAccessFlags flag)
+inline const char * getSecAccessFlagName(SecAccessFlags flag)
 {
-	switch (flag)
+    switch (flag)
```
Comment (ghalliday) [inline system/security/shared/seclib.hpp]: indent: should you change the indentation of the rest of the function to remove the tabs?

File: roxie/ccd/ccdserver.cpp [hunk 6]
```diff
@@ -12321,8 +12315,8 @@ class CRoxieServerIndexWriteActivity : public CRoxieServerInternalSinkActivity,
 
     virtual void setFileProperties(IFileDescriptor *desc) const
     {
-        IPropertyTree &partProps = desc->queryPart(0)->queryProperties(); //properties of the first file part.
-        IPropertyTree &fileProps = desc->queryProperties(); // properties of the logical file
+        //IPropertyTree &partProps = desc->queryPart(0)->queryProperties(); //properties of the first file part.
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp]: remove

**Review by ghalliday — COMMENTED** [review]
@richardkchapman a few minor changes

**Comment by richardkchapman** [comment]
@ghalliday Fixed and repushed

File: roxie/ccd/ccdfile.cpp [hunk 7]
```diff
@@ -1900,17 +1900,15 @@ class CResolvedFile : implements IResolvedFileCreator, implements ISDSSubscripti
     IMPLEMENT_IINTERFACE;
 
     CResolvedFile(const char *_lfn, const char *_physicalName, IDistributedFile *_dFile, RoxieFileType _fileType, IRoxieDaliHelper* _daliHelper, bool isDynamic, bool cacheIt, bool writeAccess, bool _isSuperFile)
-    : daliHelper(_daliHelper), lfn(_lfn), physicalName(_physicalName), dFile(_dFile), fileType(_fileType), isSuper(_isSuperFile)
+    : lfn(_lfn), physicalName(_physicalName), dFile(_dFile), fileType(_fileType), isSuper(_isSuperFile), daliHelper(_daliHelper)
     {
         cached = NULL;
         fileSize = 0;
-        fileCheckSum = 0;
```
Comment (ghalliday) [inline roxie/ccd/ccdfile.cpp]: there is still a member variable which is serialized and accessed.  This doesn't look right.

**Review by ghalliday — COMMENTED** [review]
@richardkchapman one change doesn't look right

File: roxie/ccd/ccdfile.cpp (hunk 7 above)
Comment (richardkchapman) [reply roxie/ccd/ccdfile.cpp]: Good catch - fixed

File: roxie/ccd/ccdmain.cpp [hunk 8]
```diff
@@ -1120,7 +1117,7 @@ int STARTQUERY_API start_query(int argc, const char *argv[])
                     unsigned listenQueue = roxieFarm.getPropInt("@listenQueue", DEFAULT_LISTEN_QUEUE_SIZE);
                     unsigned numThreads = roxieFarm.getPropInt("@numThreads", numServerThreads);
                     unsigned port = roxieFarm.getPropInt("@port", ROXIE_SERVER_PORT);
-                    unsigned requestArrayThreads = roxieFarm.getPropInt("@requestArrayThreads", 5);
+                    //unsigned requestArrayThreads = roxieFarm.getPropInt("@requestArrayThreads", 5);
```
Comment (ghalliday) [inline roxie/ccd/ccdmain.cpp]: Should be removed.

**Review by ghalliday — COMMENTED** [review]
@richardkchapman

File: roxie/ccd/ccdmain.cpp (hunk 8 above)
Comment (richardkchapman) [reply roxie/ccd/ccdmain.cpp]: Should probably actually raise a JIRA to see why it is not being used.

**Pull request merged**